src/js/components/Clock/Clock.js

```diff
@@ -159,11 +159,20 @@ class Clock extends Component {
         if (onChange) {
           if (elements.duration) {
             onChange(
-              `P${elements.hours}H${elements.minutes}M${elements.seconds}S`,
+              `
+              P${nextElements.hours}
```

Comment:
This will introduce newlines into the ISO time format, which shouldn't be there.

Also, instead of referencing the old `nextElements`, I think we should be referring to the latest ones by re-getting them from the latest state in the callback, perhaps?:

```
this.setState({ elements: nextElements }, () => {
  if (onChange) {
    const { elements: e2 } = this.state;
    if (elements.duration) {
      onChange(`P${e2.hours}H${e2.minutes}M${e2.seconds}S`);
    } else {
      onChange(`T${e2.hours}:${e2.minutes}:${e2.seconds}`);
    }
  }
});
```

Reply:
Nice! I'll update it. Thanks!